src/storage/src/storage/read_object.rs

```diff
@@ -414,31 +425,68 @@ impl ReadObjectResponse {
 
     #[cfg(feature = "unstable-stream")]
     #[cfg_attr(docsrs, doc(cfg(feature = "unstable-stream")))]
-    /// Convert the response to a [futures::Stream].
-    pub fn into_stream(self) -> impl futures::Stream<Item = Result<bytes::Bytes>> {
-        // TODO(#2049): implement checksum checks for streams.
+    /// Convert the response to a [Stream].
+    pub fn into_stream(self) -> impl Stream<Item = Result<bytes::Bytes>> {
```

Comment:
non-blocking comment / food for thought: we should be able to either

A: write this in terms of `next()`

https://github.com/googleapis/google-cloud-rust/blob/7c71010b19efd23b72fdad0ed087c140413ea5e9/src/lro/src/internal.rs#L326-L341

or B: hold a stream, and write `next()` in terms of the stream:

https://github.com/googleapis/google-cloud-rust/blob/7c71010b19efd23b72fdad0ed087c140413ea5e9/src/gax/src/paginator.rs#L217-L235

I am not sure how B will work with `all_bytes` though. We'd probably have to extend the bytes manually.